zeebe/stream-platform/src/test/java/io/camunda/zeebe/stream/impl/StreamProcessorTest.java

```diff
@@ -481,6 +481,77 @@ public void shouldSetSourcePointerForFollowUpRecords() {
     assertThat(logStreamReader.next().getSourceEventPosition()).isEqualTo(firstRecordPosition);
   }
 
+  @Test
+  public void shouldSetSourcePointerCorrectlyForFollowUpRecords() {
```

Comment:
:wrench: Naming wise this is very close to the existing `shouldSetSourcePointerForFollowUpRecords`. I'd suggest to rename it to make it clearer that this is about batch processing. Perhaps something like `shouldTrackSourcePointerDuringBatchProcessing`?